packages/backend/src/prometheus.ts

```diff
@@ -37,6 +38,22 @@ export default class PrometheusMetrics {
         }
     }
 
+    public monitorQueues(schedulerClient: SchedulerClient) {
+        const { queueMonitoringFrequency, enabled, ...rest } = this.config;
+        if (enabled) {
+            const queueSizeGauge = new prometheus.Gauge({
+                name: 'queue_size',
+                help: 'Number of jobs in the queue',
+                ...rest,
+                collect() {},
+            });
+            setInterval(async () => {
+                const queueSize = await schedulerClient.getQueueSize();
+                queueSizeGauge.set(queueSize);
+            }, queueMonitoringFrequency || 30 * 1000); // 30 seconds
```

Comment:
If we want to get the value at the time /metrics is requested we just need to have 

```
collect() {
   return schedulerClient.getQueueSize();
},
```

If we want to have a separate timer for this, we can remove the `collect() {},`